lib/Sema/MiscDiagnostics.cpp

```diff
@@ -3460,6 +3460,21 @@ static void diagnoseUnintendedOptionalBehavior(TypeChecker &TC, const Expr *E,
                   /* from */ srcType, /* to */ destType)
         .highlight(subExpr->getSourceRange());
 
+      if (auto decl = getDeclForExpr(subExpr)) {
+        auto kind = [&](ValueDecl *decl) -> unsigned {
```

Comment:
Instead of doing this use `Decl::getDescriptiveKind()` in the diagnostic.

Reply:
The problem with that is the diagnostics isn't as nice - you would get, for example, `implicitly unwrapped let declared here` vs. `implicitly unwrapped variable 'fk' declared here`. Rather than using `let`, `var`, `func`, it might be better to use the full name and combine let/var into `variable`. WDTY?

Reply:
I think it's okay to use `let` here, I'd prefer we stick to the descriptive kind instead of trying to invent our own here. Actually we might not even need "implicitly unwrapped" bit here since it's going to be attached to an error that already has that mentioned.